src/store/modules/order/actions.ts

```diff
@@ -10,8 +10,110 @@ import emitter from '@/event-bus'
 import store from "@/store";
 import { prepareOrderQuery } from "@/utils/solrHelper";
 import logger from "@/logger";
+import { getOrderCategory } from "@/utils/order";
 
 const actions: ActionTree<OrderState , RootState> ={
+
+  async getOrderDetails({commit}, payload ) {
+    // Show loader only when new query and not the infinite scroll
+    // if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    let resp;
+    const orderQueryPayload = prepareOrderQuery({
+      ...payload,
+      orderStatusId: 'ORDER_APPROVED',
+      orderTypeId: 'SALES_ORDER',
+      '-fulfillmentStatus': '(Cancelled OR Rejected)',
+    })
+
+    try {
+      resp = await OrderService.getOpenOrders(orderQueryPayload);
+      if (resp.status === 200 && !hasError(resp) && resp.data.grouped?.orderId?.ngroups > 0) {
+        const orderIds = resp.data.grouped?.orderId?.groups.map((order: any) => order.doclist.docs[0].orderId);
+        const { productId, ...params } = payload;
+        this.dispatch('order/getOtherOrderItem', {...params, orderIds});
+        emitter.emit("dismissLoader");
```

Comment:
We should only dismiss the loader once and not add the same emitter on multiple places. Also we have not initiated call to presentLoader on any place, so why we have added dismissLoader?